From f1c316430713ae7d526f8abc1621b09958aef281 Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd@kuhls.net>
Date: Mon, 21 Sep 2026 22:06:06 +0200
Subject: [PATCH] Fix building on Pango 1.58.0

This version added the autoptr macros which we no longer need to define.

Upstream: https://github.com/flutter/flutter/pull/193119

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 .../flutter/shell/platform/linux/fl_accessible_text_field.cc  | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/engine/src/flutter/shell/platform/linux/fl_accessible_text_field.cc b/engine/src/flutter/shell/platform/linux/fl_accessible_text_field.cc
index e99b123bbfd..f7955a32f07 100644
--- a/engine/src/flutter/shell/platform/linux/fl_accessible_text_field.cc
+++ b/engine/src/flutter/shell/platform/linux/fl_accessible_text_field.cc
@@ -6,7 +6,11 @@
 #include "flutter/shell/platform/linux/public/flutter_linux/fl_standard_message_codec.h"
 #include "flutter/shell/platform/linux/public/flutter_linux/fl_value.h"
 
+// PangoContext g_autoptr macro weren't added until 1.58.0. Add them manually.
+// https://gitlab.gnome.org/GNOME/pango/-/commit/223b147
+#if !PANGO_VERSION_CHECK(1, 58, 0)
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(PangoContext, g_object_unref)
+#endif
 // PangoLayout g_autoptr macro weren't added until 1.49.4. Add them manually.
 // https://gitlab.gnome.org/GNOME/pango/-/commit/0b84e14
 #if !PANGO_VERSION_CHECK(1, 49, 4)
-- 
2.47.3

